lib/kcerts/keys.go

```diff
@@ -0,0 +1,158 @@
+package kcerts
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"golang.org/x/crypto/ssh"
+	"strings"
+)
+
+type PrivateKey struct {
```

Comment:
This encapsulation of PrivateKeyProvider... seems a bit complex/overkill? I think I understand why you do it, but there's a single use case of it.

Personally, I would eliminate this `PrivateKey` struct definition, and let the [one caller here](https://github.com/enfabrica/enkit/pull/75/files#diff-c5353184ccf4457a5a29aaa79afa329fd3d795f1a475f55806d070b80646040bR208), just call NewSSHSigner.

(I think it would make the code easier to understand both for those touching the library, and users of Signer, who can clearly see the need of conversion)

Reply:
Changed to be more functional